src/ContactBundle.ts

```diff
@@ -21,44 +20,44 @@ export default class ContactBundle implements contact.ContactBundleV1 {
       v2: undefined,
     }).finish()
   }
+}
 
-  static fromBytes(bytes: Uint8Array): ContactBundle {
-    const bundle = this.decodeV1(bytes)
-
-    if (!bundle) {
-      throw new Error('could not parse bundle')
-    }
+export class ContactBundleV2 implements contact.ContactBundleV2 {
+  keyBundle: SignedPublicKeyBundle
 
-    if (!bundle.identityKey) {
+  constructor(bundle: contact.ContactBundleV2) {
+    if (!bundle.keyBundle) {
       throw new Error('missing keyBundle')
     }
-    if (!bundle.preKey) {
-      throw new Error('missing pre-key')
-    }
-    return new ContactBundle(
-      new PublicKeyBundle(
-        new PublicKey(bundle.identityKey),
-        new PublicKey(bundle.preKey)
-      )
-    )
+    this.keyBundle = new SignedPublicKeyBundle(bundle.keyBundle)
   }
 
-  static decodeV1(bytes: Uint8Array): publicKey.PublicKeyBundle | undefined {
-    try {
-      const b = contact.ContactBundle.decode(bytes)
-      return b.v1?.keyBundle
-    } catch (e) {
-      if (
-        e instanceof RangeError ||
-        (e instanceof Error && e.message.startsWith('invalid wire type'))
-      ) {
-        // Adds a default fallback for older versions of the proto (Which may also fail)
-        try {
-          return publicKey.PublicKeyBundle.decode(bytes)
-        } catch (e) {
-          throw new Error("Couldn't decode contact bundle: " + e)
-        }
-      }
-    }
+  toBytes(): Uint8Array {
+    return contact.ContactBundle.encode({
+      v1: undefined,
+      v2: {
+        keyBundle: this.keyBundle,
+      },
+    }).finish()
+  }
+}
+
+type ContactBundle = ContactBundleV1 | ContactBundleV2
+export default ContactBundle
```

Comment:
I'm not 100% sure how this is going to work out, but it would be nice if it did I think.

Reply:
That'll work for the type, but not the implementation classes (which is really what you care about)

Reply:
In other words, your default export here is a type instead of a class, which won't be very helpful when someone tries to import it

Reply:
This seems to me to be the most sensible definition of what `ContactBundle` should be. I don't see a sensible way to make it a class. Do you agree with that?

I have no argument why exporting this type is useful, but I note that we do have examples of that. Bunch of the proto things we import are interfaces, so exporting types shouldn't be completely useless. Would refrain from exporting it for now, until it proves it could be useful?

Reply:
Exporting it seems fine, but I think the default export from a module should be something concrete. It's kinda an implicit assumption developers have that a default export is something you can actually use.

Reply:
No default export seems fine